### ISSUE TITLE
MHN pauses directly after initdatabase.py and never completes

### ISSUE THREAD

**Issue by BeanBagKing** [threatstream/mhn#607]
TL;DR - Issues downloading emerging.rules cause the program to simply wait, and never resume.

I don't think this is a problem with MHN directly, except maybe in the way the problem is handled. I've been having problems installing MHN recently, it always froze directly after initdatabase.py. After running in debugging mode, I could see it stopping directly after this line:

`<datetime> - /opt/mhn/server/mhn/tasks/rules.py - Downloading from "http://rules.emergingthreats.net/open/snort-2.9.0/emerging.rules.tar.gz".`

At the same time, I received an alert on our IPS for "TDS Sutra - page redirecting to a SutraTDS" 
`Snort rule: alert tcp any $HTTP_PORTS -> any any (msg:"HTTP: TDS Sutra - page redirecting to a SutraTDS"; content:"?igc.ni/"; classtype:trojan-activity;) `
The odd thing is that browser download failed once, and worked a second time behind the same IPS. Wget worked the first time on the MHN server, A dozen or so MHN installs have all failed though. I'm not sure what method MHN is using to attempt the download.

I'm not familiar with the TDS Sutra rule, so I don't know why this is being triggered. 

My only suggestion for MHN might be some kind of timeout, otherwise everything just kind of waits.... And if there is something odd happening on the MHN side to trigger that alert, perhaps change the download method.

Otherwise, this can be closed. I wanted to document it, if for no other reason than to help anyone else that might encounter the same issue in the future.

**Comment by BeanBagKing** [comment]
Update: This morning I changed http://rules.emergingthreats.net/open/snort-2.9.0/emerging.rules.tar.gz to http**s**://rules.emergingthreats.net/[..] in _/opt/mhn/server/config.py_ and everything seems to be working. No more IPS block, and no more never ending wait.

**Comment by d1str0** [comment]
We should be pulling from https anyways. :/

Will fix shortly. 

Cheers

Brady Sullivan

**Comment by BeanBagKing** [comment]
I believe I found another one. While deploying a Suricata sensor I hit the same type of issue (process just seemed to hang) with an IPS alert (though not exactly the same one) directly after the following line:

/usr/bin/wget -qO - http://rules.emergingthreats.net/open/suricata-2.0/emerging.rules.tar.gz | tar -x -z -C "/opt/suricata/etc/suricata/" -f -

This appears to be part of https://github.com/threatstream/suricata that is pulled during installation. I see that URI appear in two places in "Makefile.am". 

After manually fetching/editing suricata and making necessary adjustments to the deploy script to not overwrite the corrected version, everything seems to work.

**Comment by d1str0** [comment]
Thank you for noticing and reporting! I will be pushing updates for all of these today!

http to https is a dead easy fix and it really provides a lot. Please let me know if you notice anything else similar. 

Cheers

Brady Sullivan

**Comment by d1str0** [comment]
Should be updated.
